### PR TITLE
Improve Graphite name sanitization

### PR DESCRIPTION
**Pull request by izeye**
This PR improves Graphite name sanitization by:

- sanitizing tag values from tagsAsPrefix.
- sanitizing spaces from meter names.
- sanitizing dots from tag keys and values.

Closes gh-1465

### CODE REVIEW

File: implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteNamingConvention.java [hunk 1]
```diff
@@ -33,7 +33,8 @@
      * A list that probably is blacklisted: https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/render/grammar.py#L48-L55.
      * Empirically, we have found others.
      */
-    private static final Pattern blacklistedChars = Pattern.compile("[{}(),=\\[\\]/]");
+    private static final Pattern PATTERN_NAME_BLACKLISTED_CHARS = Pattern.compile("[{}(),=\\[\\]/ ]");
+    private static final Pattern PATTERN_TAG_BLACKLISTED_CHARS = Pattern.compile("[{}(),=\\[\\]/ .]");
```
Comment (msmsimondean) [inline implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteNamingConvention.java]: @izeye  would it be possible to sanitize `?` question marks in names and tags too? Spring Boot puts question marks in metric names in some situations. Here's an example:

`httpClientRequests.clientName.some.host.name.method.POST.status.200.uri._path?key__clientKey_.count`

In this example, the question mark comes from a URI that's embedded in the metric path via a tag.

Comment (izeye) [reply implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteNamingConvention.java]: @msmsimondean Thanks for the info! Is there any reference on it like their doc or similar?

Comment (msmsimondean) [reply implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteNamingConvention.java]: @izeye here's the docs for Spring Boot: https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-metrics.html#production-ready-metrics-http-clients.  The question mark can appear in the URI tag.

Comment (izeye) [reply implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteNamingConvention.java]: @msmsimondean Sorry, my question doesn't seem to be clear. I meant to ask the reference for the following comment in https://github.com/micrometer-metrics/micrometer/issues/1465#issuecomment-508806231:

> If it helps, here are the characters that HostedGraphite (as an example of a hosted Graphite provider) allow in a Graphite metric path: 
> 
> ` a-zA-Z0-9:\._$%#=\-\[\] `

Comment (msmsimondean) [reply implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteNamingConvention.java]: Hi @izeye.  The list of characters comes from an email conversation with a HostedGraphite support agent.  The email conversation was only 3 days ago, so info should be up-to-date.  I've done a quick web search and I can't find a page on the hostedgraphite.com site that shows the list of characters.

Comment (izeye) [reply implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteNamingConvention.java]: @msmsimondean Thanks for the quick feedback! I'm not a regular Graphite user, so I couldn't judge that sanitizing "?" (question mark) is a good idea without a reference. I can update this PR accordingly if anyone can convince me on it or we can discuss on it in a separate issue.

Comment (tens91) [reply implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteNamingConvention.java]: Hi @izeye , I'm the agent at hosted graphite who @msmsimondean had been emailing. The information they've supplied is correct, those characters are the ones which we accept in graphite names. My colleague also added more information here https://github.com/micrometer-metrics/micrometer/issues/1465#issuecomment-509169319

If documentation is needed for you to update the PR, I can add this to our docs at https://hostedgraphite.com/docs - let me know.

Comment (izeye) [reply implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteNamingConvention.java]: @tens91 Thanks for the feedback! Based on the comment from @shevyf I pushed 3ffcf9d to be considered. IMHO it'd be good for users to have the info in the doc regardless of the change.

**Pull request merged**